core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -111,6 +117,11 @@ static TableMetadata newTableMetadata(Schema schema,
     int freshSortOrderId = sortOrder.isUnsorted() ? sortOrder.orderId() : INITIAL_SORT_ORDER_ID;
     SortOrder freshSortOrder = freshSortOrder(freshSortOrderId, freshSchema, sortOrder);
 
+    // rebuild the row identifier using the new column ids
+    int freshRowIdentifierVersion = rowIdentifier.isNotIdentified() ?
+        rowIdentifier.rowIdVersion() : INITIAL_ROW_IDENTIFIER_VERSION;
```

Comment:
Yes, because not identified is default that has ID 0. Otherwise the row key should have ID 1 in this method, and then get merged with current metadata in create/replace table transaction.